gpii/node_modules/flowManager/src/PSPChannel.js

```diff
@@ -300,3 +323,26 @@ gpii.pspChannel.receiveMessage = function (message, pspChannel) {
     }
     pspChannel.applier.change(message.path, message.value, "ADD", "PSP");
 };
+
+gpii.pspChannel.generateDefaultSettingControls = function (defaultSettings, schemas) {
+    var defaultSettingControls = {};
+    var defaultPreferences = fluid.get(defaultSettings, ["contexts", "gpii-default", "preferences"]);
+    fluid.each(defaultPreferences, function (prefsVal, prefsKey) {
+        gpii.pspChannel.emitSettingControl(defaultSettingControls, schemas[prefsKey], prefsVal, [prefsKey], "live");
```

Comment:
Note that this liveness annotation isn't actually correct - it should be looked up in the SR, but in practice the system doesn't support any non-live solutions.  This is the kind of thing that leads me to suspect that this strategy here is not right and that we should actually be reading the "defaultLifecycleInstructions" listed here: https://github.com/GPII/universal/blob/89187a28c7a06c82c8aa26178b833f64f05fefdb/gpii/node_modules/flowManager/src/FlowManager.js#L284

Note further that the "defaultLifecycleInstructions" itself is incorrect since it assigns into the component's options which is not permitted.

Reply:
It may well not be worth correcting either of these problems in the time we have available, we could just JIRA them and look at it when you are back

Reply:
Do you mean these internally generated values should assign to member options rather than options?

Reply:
Moving "defaultLifecycleInstructions" out into a member option won't be able to distribute its value to subcomponents: https://github.com/GPII/universal/blob/89187a28c7a06c82c8aa26178b833f64f05fefdb/gpii/node_modules/flowManager/src/FlowManager.js#L129

When options block become immutable, is there a plan to allow distributeOptions "source" element to point to a member option?

Reply:
You can distribute from members simply by distributing from their resultant site in the component - that is, {that}.thing . But before we make options immutable, we will already have the improvements from the current framework update which will allow you to express this material as an expander in the component's options even though it requires I/O.

Reply:
There might be problem with getting live solutions from "defaultLifecycleInstructions" whose structure looks like [this](https://pastebin.com/xbd6DBfB) and has no trace of the original setting term "http://registry.gpii.net/common/cursorSize" that is required as [the "fullKey" value](https://github.com/GPII/universal/blob/master/gpii/node_modules/flowManager/src/PSPChannel.js#L158) at generating settingControls block.